### ISSUE TITLE
Open source the Microsoft.Maps.Unity package

### ISSUE THREAD

**Issue by xwipeoutx** [microsoft/MapsSDK-Unity#43]
**Is your feature request related to a problem? Please describe.**
The rest of MixedReality Unity implementations are open source - this a request to do the same with Microsoft.Maps.Unity, which is currently a NuGet containing an obfuscated DLL.

I'm having to reverse engineer a lot of things to add features I want - like hill/relief shading, exaggerated elevation.  Even the shader variable names are obfuscated!

Additionally, I can't contribute this stuff back, as it's all such a hack.

**Describe the solution you'd like**
Please consider open sourcing the base `Microsoft.Maps.Unity` assembly.  This will keep it in line with the rest of the Microsoft MR Unity ecosystem (e.g. the MRTK)

**Describe alternatives you've considered**
I've decompiled, even writing a [fiddle](https://dotnetfiddle.net/z2UZsw) to extract the strings from byte arrays so I can understand the shaders better.

**Additional context**
Just want to say this map control is really nice - smoother, better blending, good architecture and documentation.  Blows alternatives I've tried out of the water.  There are a lot of clients that want to see this rich 3D map on a HoloLens, and it's being held back unnecessarily by this closed source approach.  Please follow the lead of MRTK (not to mention .NET) and ❤️  open source.

**Comment by kircher1** [comment]
The portions of the SDK that are currently OSS live under the [supporting scripts](https://github.com/microsoft/MapsSDK-Unity/tree/master/SupportingScripts). Contributions are welcome there! These scripts and shaders are available directly in the NuGet package too (not built into the DLL).

Shaders (or the material, rather) can be overriden in the MapRenderer's editor UI, but if you go down that route, it is best to base this off of the existing shaders as a starting point. The [ContourLinesLayer](https://github.com/microsoft/MapsSDK-Unity/blob/master/SupportingScripts/Scripts/MapContourLineLayer.cs) provides an example of how you can add custom content to the existing shader.

In summary, we are making what pieces possible open source, and will continue to add content to the OSS directories as we progress with the SDK. There is nothing specific to share now, but the request of moving more functionality to OSS is noted!

**Comment by xwipeoutx** [comment]
Thanks for your speedy reply - here's a bit more context for the desire for open sourcing the upstream stuff - I'm keen to understand the motivations behind a bunch of it being closed source - why are some pieces "possible" and others not?

Your suggestions are I've been doing so far - but it's like an iceberg - the OSS part is fairly minor, mostly helper scripts on the surface, and the majority of it is hidden. It feels antithetical to the way the the Mixed Reality support works, and is quite a slowdown.

Unfortunately, it's not possible to contribute back what I've done, since a lot of the logic for how the materials get their keywords enabled are hidden in the map renderer.  

For example, I've implemented some relief shading:

![Relief shaded map example](https://user-images.githubusercontent.com/1008567/73797924-08e8e480-47fd-11ea-8ea8-ddd71a0a13fa.png)

But it's dependent on the `ENABLE_ELEVATION_TEXTURE` keyword (and presence of the elevation texture), which is controlled by the `MapRenderer`, and not exposed in code.  I also can't control the LOD for the elevation texture separately from the imagery, so the effort required for me to have higher-quality relief maps rises dramatically (have to load my relief maps as a separate tile layer or something - arguably, I should be pregenerating these normal maps anyway for better performance 😂)

In any case, it's all _doable_, but a major drag on productivity, and not something I can contribute back - and that's just one example.

**Comment by REscanDon** [comment]
Resolving this issue as closed an hour after it was opened definitely does not inspire confidence in this project. The Mapbox Unity plugin is arguably the closest competitor to this effort and it's 100% open source. Those of us outside Microsoft have zero clarity on the size of the team supporting this effort or the likely survival rate of a garage-initiated project like this. Keeping a project like this closed source as it is feels very old days of Microsoft in addition to feeling entirely counterproductive given that any monetization of the effort will almost certainly be through the bing maps API not through this plugin directly.

**Comment by ketourneau** [comment]
@kircher1 
It would indeed be interesting to make opensource the core Microsoft.Maps.Unity package.
It's an important argument when we have to choose between several package in a company.
